src/io/file_descriptor.cr

```diff
@@ -11,7 +11,12 @@ class IO::FileDescriptor < IO
   getter fd
 
   def initialize(@fd, blocking = false)
-    @closed = false
+    @closed =
+      {% unless flag?(:win32) %}
+        !fcntl(LibC::F_GETFD) rescue true
+      {% else %}
+        false
+      {% end %}
```

Comment:
perhaps we should define a `system_closed?` and check that here and after reopen.

Reply:
No need to check after reopen since it has to be open if dup succeeded. Still made the method though, it looks nicer.